bicytok/distanceMetricFuncs.py

```diff
@@ -1,641 +1,221 @@
-                    / markAvg
-                )
-            # Compare to all non-Tregs
-            elif offTargState == 1:
-                offTargCellMark = (
-                    CITE_DF.loc[CITE_DF["CellType2"] != "Treg"][marker].values / markAvg
-                )
-            # Compare to naive Tregs
-            elif offTargState == 2:
-                offTargCellMark = (
-                    CITE_DF.loc[CITE_DF["CellType3"] == "Treg Naive"][marker].values
-                    / markAvg
-                )
-            if np.mean(targCellMark) > np.mean(offTargCellMark):
-                kdeTarg = KernelDensity(kernel="gaussian").fit(
-                    targCellMark.reshape(-1, 1)
-                )
-                kdeOffTarg = KernelDensity(kernel="gaussian").fit(
-                    offTargCellMark.reshape(-1, 1)
-                )
-                minVal = np.minimum(targCellMark.min(), offTargCellMark.min()) - 10
-                maxVal = np.maximum(targCellMark.max(), offTargCellMark.max()) + 10
-                outcomes = np.arange(minVal, maxVal + 1).reshape(-1, 1)
-                distTarg = np.exp(kdeTarg.score_samples(outcomes))
-                distOffTarg = np.exp(kdeOffTarg.score_samples(outcomes))
-                KL_div = stats.entropy(
-                    distOffTarg.flatten() + 1e-200, distTarg.flatten() + 1e-200, base=2
-                )
-                markerDF = pd.concat(
-                    [
-                        markerDF,
-                        pd.DataFrame(
-                            {
-                                "Marker": [marker],
-                                "Wasserstein Distance": stats.wasserstein_distance(
-                                    targCellMark, offTargCellMark
-                                ),
-                                "KL Divergence": KL_div,
-                            }
-                        ),
-                    ]
-                )
 
-    corrsDF = pd.DataFrame()
-    for i, distance in enumerate(["Wasserstein Distance", "KL Divergence"]):
-        ratioDF = markerDF.sort_values(by=distance)
-        posCorrs = ratioDF.tail(numFactors).Marker.values
-        corrsDF = pd.concat(
-            [corrsDF, pd.DataFrame({"Distance": distance, "Marker": posCorrs})]
-        )
-        markerDF = markerDF.loc[markerDF["Marker"].isin(posCorrs)]
-        sns.barplot(
-            data=ratioDF.tail(numFactors), y="Marker", x=distance, ax=ax[i], color="k"
-        )
-        ax[i].set(xscale="log")
-        ax[0].set(title="Wasserstein Distance - Surface Markers")
-        ax[1].set(title="KL Divergence - Surface Markers")
-    return corrsDF
-
-
-def get_conversion_factor(weightDF, receptor_name):
+    recAvg = np.mean(recAbun)
+
+    assert recAvg != 0
+
+    targAbun = recAbun[targ] / recAvg
+    offTargAbun = recAbun[offTarg] / recAvg
+
+    return targAbun, offTargAbun
+
+
+def calculateKLDiv(targAbun, offTargAbun, dims):
```

Comment:
What's the difference between this and the KL output of `KL_EMD_1D()`?

Reply:
There is no difference. I am hoping to make this function treat all dimensions the same to simplify the other functions.